extensions/opencv/CaptureRTSPFrame.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CaptureRTSPFrame.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+static core::Property rtspUsername;
+static core::Property rtspPassword;
+static core::Property rtspHostname;
+static core::Property rtspURI;
+static core::Property captureFrameRate;
+static core::Property imageEncoding;
+
+core::Property CaptureRTSPFrame::RTSPUsername(  // NOLINT
+    "RTSP Username",
+    "The username for connecting to the RTSP stream", "");
+core::Property CaptureRTSPFrame::RTSPPassword(  // NOLINT
+    "RTSP Password",
+    "Password used to connect to the RTSP stream", "");
+core::Property CaptureRTSPFrame::RTSPHostname(  // NOLINT
+    "RTSP Hostname",
+    "Hostname of the RTSP stream we are trying to connect to", "");
+core::Property CaptureRTSPFrame::RTSPURI(  // NOLINT
+    "RTSP URI",
+    "URI that should be appended to the RTSP stream hostname", "");
+core::Property CaptureRTSPFrame::RTSPPort(  // NOLINT
+    "RTSP Port",
+    "Port that should be connected to to receive RTSP Frames",
+    "");
+core::Property CaptureRTSPFrame::ImageEncoding( // NOLINT
```

Comment:
Can we switch this to using the builder? This will allow us to select a small set of allowable values for image encodings. 

https://github.com/apache/nifi-minifi-cpp/blob/master/extensions/civetweb/processors/ListenHTTP.cpp#L47

Is an example. May help with the others above, especially for port.